pilot/pkg/xds/debuggen.go

```diff
@@ -83,31 +83,80 @@ func NewDebugGen(s *DiscoveryServer, systemNamespace string, debugMux *http.Serv
 
 // Generate XDS debug responses according to the incoming debug request
 func (dg *DebugGen) Generate(proxy *model.Proxy, w *model.WatchedResource, req *model.PushRequest) (model.Resources, model.XdsLogDetails, error) {
-	res := model.Resources{}
-	var buffer bytes.Buffer
+	if err := validateProxyAuthentication(proxy, w); err != nil {
+		return nil, model.DefaultXdsLogDetails, err
+	}
+
+	resourceName, err := parseAndValidateDebugRequest(proxy, w, dg)
+	if err != nil {
+		return nil, model.DefaultXdsLogDetails, err
+	}
+
+	buffer := processDebugRequest(dg, resourceName)
+
+	res := model.Resources{&discovery.Resource{
+		Name: resourceName,
+		Resource: &anypb.Any{
+			TypeUrl: v3.DebugType,
+			Value:   buffer.Bytes(),
+		},
+	}}
+	return res, model.DefaultXdsLogDetails, nil
+}
+
+// GenerateDeltas XDS debug responses according to the incoming debug request
+func (dg *DebugGen) GenerateDeltas(
```

Comment:
any diff with `Generate`

Reply:
No diff as the resources retrived are the same right now.